src/Dialogs/Device/DeviceEditWidget.cpp

```diff
@@ -443,7 +468,7 @@ void
 DeviceEditWidget::OnModified(DataField &df) noexcept
 {
   if (IsDataField(Port, df) || IsDataField(Driver, df) ||
-      IsDataField(UseSecondDriver, df) || IsDataField(K6Bt, df))
+      IsDataField(UseSecondDriver, df) || IsDataField(K6Bt, df) || IsDataField(EngineTypes, df))
```

Comment:
What visibilities change when the engine type is modified?

Reply:
I don't get the question. I thought this will update the drop down list, it does not?

Reply:
No, it does not. But even if it would - why do you want to update the drop down list? The drop down list of Engine Type is static, it never needs to be updated.
Look at the code, this calls `UpdateVisibilities()`; it makes certain rows visible or invisible, depending on certain selected features. If you enable K6Bt mode, the "baud rate" row must be shown; if you disable K6Bt, it might be hidden.